perfkitbenchmarker/virtual_machine.py

```diff
@@ -285,8 +314,30 @@ def RemoteCopy(self, file_path, remote_path='', copy_to=True):
                     (retcode, full_cmd, stdout, stderr))
       raise errors.VmUtil.SshConnectionError(error_text)
 
-  def LongRunningRemoteCommand(self, command):
-    """Runs a long running command on the VM in a robust way.
+  def _PushAsyncCommandScripts(self):
+    """Pushes the scripts required by AsyncRemoteCommand to this VM.
+
+    If the scripts have already been placed on the VM, this is a noop.
+    """
+    with self._has_remote_command_script_lock:
+      if not self._has_remote_command_script:
+        for f in (EXECUTE_COMMAND, WAIT_FOR_COMMAND):
+          self.PushDataFile(f, os.path.join(vm_util.VM_TMP_DIR,
+                                            os.path.basename(f)))
+        self._has_remote_command_script = True
+
+  def AsyncRemoteCommand(self, command):
```

Comment:
I think "RobustRemoteCommand" might be a better name than AsyncRemoteCommand. Having Async*** starting a method name gives me an immediate impression that this API will return immediately to the caller when called, and starts doing its work asynchronously, and it would be the *caller*'s responsibility to poll or wait for the completion of this work. But that's not the case here, this method is still a synchronous method, it will start two concurrent commands, and wait for the commands to finish before it returns control to the caller.

The intention of this method is to provide a better, and more robust way of executing a remote command and deliver the results back to the caller safely. It's really just a better implementation of the original RemoteCommand() method (with some extra overhead). We can also name it as RemoteCommandV2() or something like that. (if you don't like the "RobustRemoteCommand()" suggestion. :-)